### ISSUE TITLE
Composer

### ISSUE THREAD

**Issue by yani** [yelirekim/introspective#3]
Would it be possible to do this without Composer?

The reason I came across this package in the first place was to remove a need for Composer.

**Comment by Blackskyliner** [comment]
No, this package is more for introspection of an codebase, so you get correct context sensitive code completion. To cite the README.md

```
The main driver behind features is supplying this information to code completion plugins for various editors.
```

Kinda issue on-topic but repository off-topic, why won't you use composer? I mean it is somehow "the thing" to get a somehow good dependency management going in a PHP project. it would be like programming in JS with node and not using npm to install dependencies... Or like using ruby without gem/bundler...

And if you want tu use this package w/o composer, then go ahead and just download that archive as an zip, nobody stops you from manually adding this to your project. But I guess I can then also say, welcome back in the manually managed dependency hell. :-)

**Comment by yani** [comment]
I'm working on a big project where dependencies shouldn't be used because of filesize, security and licenses. I love manually managing all my dependencies as it allows me to have a full understanding of what's happening, without trusting a third party.

I believe that composer and frameworks are very good for websites, but for actual software I find them way too excessive. And that's what I'm trying to do here, to make software instead of websites.

Thanks for your comment though. I'll have a look and see if I can get it working without composer.

**Comment by Blackskyliner** [comment]
You could use composer and have those traits you describe even for all your NIH/home-brew stuff.
Just use your own Composer Repository Server, like Satis. There you could also, after an possible audit, add external, already composer compatible packages.

I also know some of the downsides of composer, like vanishing repositories, making a deployment impossible. (already happened often enough to me) So I get some points you make there.

But if you are really doing "Software" instead of "Websites" why even bother with PHP then and not using some other language that fits better to those need you may have? I mean PHP is _the_ definition of dynamic websites and it's mostly developed for that entire purpose. (Not even to mention that every computer program is "Software" even a "Website" in some terms)

Btw. especially in the category "security" I would advise you not to brew your own Packges/Classes and stuff. I mean it's good to know what happens, but my guess would be you are not a crypto/security expert and using an external package there will be a good thing to do ;-) - Keeping those manually up-to-date, if you want to, no problems here it's your decision ;-) - But like mentioned you could look at the personal composer repository stuff and at least get a somewhat consistent dependency management, that new co-workers will understand and could manage, as it's more-or-less industry standard nowadays. 

Oh, and as you created the issue and I guess it's mostly answered completely, may you close the issue then?
